File: rider-location-service/src/main/java/com/potatorider/repository/RiderLocationRepository.java
package com.potatorider.repository;

import com.potatorider.domain.RiderLocation;

import reactor.core.publisher.Mono;

public interface RiderLocationRepository {

    Mono<Boolean> setIfPresent(RiderLocation riderLocation);

    Mono<Boolean> setIfAbsent(RiderLocation riderLocation);

    Mono<RiderLocation> getLocation(String locationId);
}

File: rider-location-service/src/main/java/com/potatorider/repository/RiderLocationRepository.kt
package com.potatorider.repository

import com.potatorider.domain.RiderLocation
import reactor.core.publisher.Mono

interface RiderLocationRepository {
    fun setIfPresent(riderLocation: RiderLocation): Mono<Boolean>

    fun setIfAbsent(riderLocation: RiderLocation): Mono<Boolean>

    fun getLocation(locationId: String): Mono<RiderLocation>
}
